pkg/controller/node/nodecontroller.go

```diff
@@ -482,9 +482,33 @@ func (nc *NodeController) monitorNodeStatus() error {
 						continue
 					}
 					if remaining {
-						// queue eviction of the pods on the node
+						// Immediately evict pods (skip rate-limited evictor)
 						glog.V(2).Infof("Deleting node %s is delayed while pods are evicted", node.Name)
-						nc.evictPods(node.Name)
+						go func(nodeName string) {
+							nc.evictorLock.Lock()
+							defer nc.evictorLock.Unlock()
+							remaining, err := nc.deletePods(nodeName)
+							if err != nil {
+								glog.Errorf("Unable to evict pods from node %s: %v", nodeName, err)
+								nc.podEvictor.Add(nodeName)
+								return
+							}
+							if !remaining {
+								return
+							}
+							completed, _, err := nc.terminatePods(nodeName, time.Now())
```

Comment:
In practice this method will always return completed false.  It would be better to go straight to the graceful check queue than to call terminate pods.